subconscious/model.py

```diff
@@ -101,7 +106,7 @@ def __init__(self, allow_unknown_cols=False, **kwargs):
                     )
                 setattr(self, column.name, value)
             else:
-                assert not column.required, 'Column `{}` in `{}` is required'.format(
+                assert not (column.required and not getattr(column, 'auto', False)), 'Column `{}` in `{}` is required'.format(
```

Comment:
Please use exceptions and not assertions in this library. If you rebase to the latest you'll find that there aren't asserts anymore.